spot_driver/spot_driver/spot_ros2.py

```diff
@@ -2799,17 +2800,23 @@ def step(self) -> None:
 
     def destroy_node(self) -> None:
         self.get_logger().info("Shutting down ROS driver for Spot")
-        if self.spot_wrapper is not None:
-            self.spot_wrapper.sit()
```

Comment:
@tcappellari-bdai @khughes-bdai @jbarry-bdai @baxelrod-bdai How bad would it be to send a stop command to spot here instead?

Reply:
should be ok - I can try that out today

Reply:
I added in the stop line while running the simple walk forward example and it worked but does result in some error spew in the terminal caused by the interruption as the driver shuts down. Unsure if that's something we want to do something about or if its ok to have that as expected behavior.

```
[spot_ros2-1] [INFO] [1709740914.500403850] [Betty.spot_ros2]: Shutting down ROS driver for Spot
[spot_ros2-1] Failed to publish log message to rosout: publisher's context is invalid, at ./src/rcl/publisher.c:389
[spot_ros2-1] [_channel.py:246] Exception in callback <function FutureWrapper.add_done_callback.<locals>.<lambda> at 0x7f93cc094430>: RCLError("Failed to publish: publisher's context is invalid, at ./src/rcl/publisher.c:389")
[spot_ros2-1] [_channel.py:246] Exception in callback <function FutureWrapper.add_done_callback.<locals>.<lambda> at 0x7f93cc094160>: RCLError("Failed to publish: publisher's context is invalid, at ./src/rcl/publisher.c:389")
[spot_ros2-1] [_channel.py:246] Exception in callback <function FutureWrapper.add_done_callback.<locals>.<lambda> at 0x7f940500b7f0>: RCLError("Failed to publish: publisher's context is invalid, at ./src/rcl/publisher.c:389")
[spot_ros2-1] [ERROR] [1709740914.569157846] [Betty.spot_ros2]: Error raised in execute callback: Failed to publish feedback with an action server: feedback publisher is invalid, at ./src/rcl_action/action_server.c:915
[spot_ros2-1] Failed to publish log message to rosout: publisher's context is invalid, at ./src/rcl/publisher.c:389
[spot_ros2-1] Traceback (most recent call last):
[spot_ros2-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/action/server.py", line 325, in _execute_goal
[spot_ros2-1]     execute_result = await await_or_execute(execute_callback, goal_handle)
[spot_ros2-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 107, in await_or_execute
[spot_ros2-1]     return callback(*args)
[spot_ros2-1]   File "/workspaces/bdai/_build/install/spot_driver/lib/spot_driver/spot_ros2", line 2110, in handle_robot_command
[spot_ros2-1]     goal_handle.publish_feedback(feedback_msg)
[spot_ros2-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/action/server.py", line 147, in publish_feedback
[spot_ros2-1]     self._action_server._handle.publish_feedback(feedback_message)
[spot_ros2-1] rclpy._rclpy_pybind11.RCLError: Failed to publish feedback with an action server: feedback publisher is invalid, at ./src/rcl_action/action_server.c:915
[spot_ros2-1] [WARN] [1709740914.573375257] [Betty.spot_ros2]: Goal state not set, assuming aborted. Goal ID: [173 208 182 167 116  40  78 180 156  18   1 238  49 224  19  66]
[spot_ros2-1] Failed to publish log message to rosout: publisher's context is invalid, at ./src/rcl/publisher.c:389
```

Reply:
Do we need to leave in the sitting as an option for backwards compatibility?

I would prefer that the driver not try to send anything to Spot while it's shutting down.  It's a weird time in Python and seems like it's going to lead to headaches for not a lot of gain?

Reply:
I think sending a stop command makes sense but removing the sit shouldn't cause an backwards compatibility issues (to my knowledge) unless the user specifically wants spot to sit on shutdown?

Reply:
I'm assuming that this was added by our external collaborators for a reason.  I think they may do all of their control through the driver as opposed to the more hybrid driver/tablet workflow we use.  Maybe check on #external-spot-ros2 and see if anyone cares?

I still think the less we do in a destructor the better but will bow to superior knowledge.

Reply:
The stop result I think makes sense I think? I believe the stop is interrupting before the feedback gets there from the action server. I really don't have any big issue with it.